### ISSUE TITLE
Please update for EF Core 6

### ISSUE THREAD

**Issue by wdhenrik** [dotnet/EntityFramework.Docs#3694]
I've followed the instruction in this page, but I'm getting the error "Could not load file or assembly 'Firebird.Migrations, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified."

I think I've done all the steps correct, but now I'm wondering if this approach has broken with Startup.cs being merged into Program.cs.  Can this article be updated for EF Core 6?

Thanks,
Wes

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b545eeff-6c33-3cc0-09ce-a3ad342d7438
* Version Independent ID: 34bc1bac-7338-d8ab-390d-8c5e88a48c10
* Content: [Using a Separate Migrations Project - EF Core](https://docs.microsoft.com/en-us/ef/core/managing-schemas/migrations/projects?tabs=dotnet-core-cli)
* Content Source: [entity-framework/core/managing-schemas/migrations/projects.md](https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/managing-schemas/migrations/projects.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @bricelam
* Microsoft Alias: **avickers**

**Comment by roji** [comment]
@wdhenrik can you please submit a runnable code sample showing the problem? Startup.cs and Program.cs being merged shouldn't have anything to do with this.

**Comment by wdhenrik** [comment]
What is the best way for me to provide that.  Can I upload a solution?

I'm just starting a new project to replace an old WebForms intranet site and am new to EFC.  I've only used TFS so am new to Github as well.

**Comment by roji** [comment]
The best way to provide a repro is to narrow it down to a minimal (but still runnable!) console program - at that point you can just post the source code directly into the issue (use markdown code fences, surrounding your code with ```c#).

Otherwise, you can push a minimal project as a personal repo, and link to it here.

**Comment by wdhenrik** [comment]
https://github.com/wdhenrik/EFC-SeparateMigrations

I've add the repo and invited you to it.  Thanks for taking a look.

**Comment by roji** [comment]
@wdhenrik that repo seems to be empty.

**Comment by wdhenrik** [comment]
https://github.com/wdhenrik/efc-separatemigrations

I've deleted and republished it.

**Comment by roji** [comment]
@wdhenrik your Firebird project sets EF Core up as follows:

```c#
builder.Services.AddDbContextFactory<FirebirdContext>(
    opt => opt.UseSqlServer(builder.Configuration.GetConnectionString("Firebird")
                            , x=>x.MigrationsAssembly("Firebird.Migrations")
    ));
```

This line is in the Firebird project, which doesn't reference the Migrations project - this is why the assembly isn't found and you get an error.

If you want to have a separate project for your migrations (and not reference that from your main project), you'll need to run your migrations in that migrations project.

**Comment by wdhenrik** [comment]
I received a circular reference error when trying to add a Migrations reference from the Firebird (Startup) project.  I followed the instructions in step 5 to add the BaseOutputPath to the migrations project instead.  That is when I was receiving the file not found error.

However, I have now created a new solution and separated my DbContext into a separate class library as well.  This avoids the circular reference as the Firebird (Startup) project can now reference the Migrations and the DbContext, Migrations only references the DbContext, and the DbContext library doesn’t need either.

I also changed the x.MigrationsAssembly to only use the library name “Migrations”, rather than the solution.library name “Firebird.Migrations”.

I appreciate you looking into it.

Thanks,
Wes
